wnfs-wasm/src/fs/private/file.rs

```diff
@@ -1,42 +1,87 @@
 //! The bindgen API for PrivateFile.
 
+use crate::{
+    fs::{
+        utils::{self, error},
+        BlockStore, ForeignBlockStore, JsResult, Namefilter, PrivateForest, Rng,
+    },
+    value,
+};
 use chrono::{DateTime, Utc};
-use js_sys::Date;
-use wasm_bindgen::prelude::wasm_bindgen;
+use js_sys::{Date, Promise, Uint8Array};
+use std::rc::Rc;
+use wasm_bindgen::{prelude::wasm_bindgen, JsValue};
+use wasm_bindgen_futures::future_to_promise;
 use wnfs::{Id, PrivateFile as WnfsPrivateFile};
 
-use crate::fs::{JsResult, Namefilter, Rng};
-
 //--------------------------------------------------------------------------------------------------
 // Type Definitions
 //--------------------------------------------------------------------------------------------------
 
 /// A file in a WNFS public file system.
 #[wasm_bindgen]
-pub struct PrivateFile(WnfsPrivateFile);
+pub struct PrivateFile(pub(crate) Rc<WnfsPrivateFile>);
```

Comment:
maybe I'm missing it in the examples, but does this really need to be pub(crate) inside? Maybe point me to where it's used as pub?

Reply:
In https://github.com/wnfs-wg/rs-wnfs/blob/appcypher/private-file-sharding/wnfs-wasm/src/fs/utils.rs#L81. But since it is only really used in one place I should probably remove the pub(crate) and pass in the PrivateFile itself.